src/sensor.cpp

```diff
@@ -347,14 +347,28 @@ namespace librealsense
                     int width = vsp ? vsp->get_width() : 0;
                     int height = vsp ? vsp->get_height() : 0;
 
-                    frame_holder fh = _source.alloc_frame(stream_to_frame_types(req_profile_base->get_stream_type()), width * height * bpp / 8, fr->additional_data, requires_processing);
+                    frame_holder fh = _source.alloc_frame(stream_to_frame_types(req_profile_base->get_stream_type()), width * height * (float)bpp / 8.f, fr->additional_data, requires_processing);
```

Comment:
warning!
Will this do? You may not need the float...
```
size_t bytes_needed = (w * h * bpp) >> 3;
```
You should probably assert, but we're still expecting the total to be a multiple of 8, no?

Reply:
(note that you can move the `expected_size` calculation up here and change it to the above)

Reply:
done